### PR TITLE
Enhancing UX for adding content

### LINKED ISSUE #2191: Create a consistent UX for adding content
**Is your feature request related to a problem? Please describe.**
- Please work with @Ayush0Chaudhary for comments, advice and review of your PR.

There isn't a consistent way of adding content. Each of these features uses a different methodology and in some cases multiple add buttons are visible.
1. Add post
1. Add event
1. Add chat (The icon is not available in current `develop` or `main` branches as chat does not work. We do not want this icon restored in this issue as it is a potential GSoC 2024 project idea. This information is just to provide some context)

This is visible here
- https://youtu.be/86AzK_Y_GCQ?t=206

**Describe the solution you'd like**
- A consistent way of adding content so that only one add icon is visible when an activity is selected. 

**Describe alternatives you've considered**
- The LinkedIn mobile app has one possible approach (there may be others) where a right hand drawer is created that covers the previous screen when a message needs to be sent. The `add post` icon gets replaced with an `add message` icon
- There may be other apps that have other ways of achieving this goal.

**Approach to be followed (optional)**
- N/A

**Additional context**
- You must get the test **_code coverage up to 100%_** for any edited files.

**Potential internship candidates**

- Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Dante291**
**What kind of change does this PR introduce?**

Enhancing UX for adding content so that no longer add sign button are there in a particular screen of the App.

**Issue Number:**

Fixes #2191 

**Did you add tests for your changes?**

Yes 

**Snapshots/Videos:**
Screenshot_1701688091

**Summary**

Currently there are more than one add/plus sign button on some screen of the app to add content this affects the user experience of the App. I added Floating Action Button(FAB) on the feed screen of the app so that there wouldn't be more than one add/plus button on any screen of the app.

This FAB navigates us to the add post page screen to make new posts, took reference from X/Twitter app for making new tweets.

**Have you read the [contributing guide] (https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

File: lib/constants/routing_constants.dart [hunk 1]
```diff
@@ -99,4 +99,7 @@ class Routes {
 
   /// static variable to access pinnedpostscreen.
   static const String pinnedPostScreen = '/pinnedpostscreen';
+
+  /// static variable.
```
Comment (literalEval) [inline lib/constants/routing_constants.dart:103]: Write a better doc.

File: lib/views/after_auth_screens/feed/organization_feed.dart [hunk 2]
```diff
@@ -26,6 +27,19 @@ class OrganizationFeed extends StatelessWidget {
       builder: (context, model, child) {
         print(model.posts);
```
Comment (literalEval) [inline lib/views/after_auth_screens/feed/organization_feed.dart:28]: Can you remove this print statement as a part of your PR itself ?

File: lib/views/after_auth_screens/feed/organization_feed.dart [hunk 3]
```diff
@@ -26,6 +27,19 @@ class OrganizationFeed extends StatelessWidget {
...
+          floatingActionButton: FloatingActionButton(
+            shape: const CircleBorder(side: BorderSide.none),
+            key: const Key('floating_action_btn'),
+            backgroundColor: Colors.green,
+            onPressed: () {
+              navigationService.pushScreen('/addpostscreen');
+            },
+            child: const Icon(
+              Icons.add,
+              size: 35,
```
Comment (literalEval) [inline lib/views/after_auth_screens/feed/organization_feed.dart:39]: Don't use constant sizing. Use SizeConfig instead.

**Review by literalEval — CHANGES_REQUESTED** [review]

**Comment by literalEval** [comment]
@Dante291 since it is a UI change, it'll help if you can please add before and after screen records.

**Comment by Ayush0Chaudhary** [comment]
@Dante291 add code coverge ss too

**Review by Ayush0Chaudhary — COMMENTED** [review]
Make sure you cover all the lines.

**Comment by Dante291** [comment]
Screenshot_2023-12-05-23-47-44-236_com example talawa

**Comment by Dante291** [comment]
image


Added test for all the lines I have added/edited.

**Comment by Dante291** [comment]
> Don't use constant sizing. Use SizeConfig instead.

@literalEval I tried using SizeConfig for setting up the size of the icon but it showed me error, Invalid constant value, I also noticed constant sizing is used for setting up the size for Icons throughout the app.

Other than this I have resolved all the above asked changes.

**Comment by noman2002** [comment]
> > Don't use constant sizing. Use SizeConfig instead.
> 
> @literalEval I tried using SizeConfig for setting up the size of the icon but it showed me error, Invalid constant value, I also noticed constant sizing is used for setting up the size for Icons throughout the app.
> 
> Other than this I have resolved all the above asked changes.

Remove const from the widget. You'll not the error.

**Comment by Dante291** [comment]
> > > Don't use constant sizing. Use SizeConfig instead.
> > 
> > 
> > @literalEval I tried using SizeConfig for setting up the size of the icon but it showed me error, Invalid constant value, I also noticed constant sizing is used for setting up the size for Icons throughout the app.
> > Other than this I have resolved all the above asked changes.
> 
> Remove const from the widget. You'll not get the error.

@noman2002  I didn't notice that, Thanks sir it helped. Please review or if you have any other UI or any changes do lemme know.

**Pull request merged**